times_miro.gms

```diff
@@ -361,191 +362,254 @@ os.environ['CUBEINPUTDOM'] = 'siName,typ,dd,' + ','.join(list(gams.get('xidom'))
 dd = []
 offeps = []
 for df in glob.glob(r'%DDPREFIX% '.rstrip()+'*.dd'):
-   if not '_ts.dd' in df:
+    if df[len(r'%DDPREFIX% '.rstrip()):].strip() != isTS:
      ddbase = os.path.splitext(os.path.basename(df))[0]
      s = 'grep -i offeps "' + df + '" > ' + os.devnull
      rc = os.system(s)
      if 0 == rc:
        offeps.append(ddbase)
+       #add offeps information to scenddmap
+       for idx, (order,ddName,offeps) in enumerate(scenddmap):
+         if ddName.lower() == ddbase.lower():
+           scenddmap[idx][2] = 'true'
      dd.append(ddbase)
 gams.set('dd',dd)
-gams.set('offeps',offeps)
 db['MILESTONYR'].copy_symbol(gams.db['MILESTONYR'])
-gams.set('gmsRunScenario',[run_name])
+scenddmap = [tuple(l) for l in scenddmap]
+#TODO: mergeType=MergeType.REPLACE?
 gams.set('scenddmap',scenddmap)
 # process myrun.lst for compile time variables
+print("### Process myrun.lst for compile time variables")
 with open('myrun.lst') as flst:
   rl = flst.readlines()
 start = [i for i, s in enumerate(rl) if 'Level SetVal' in s][0]+2
-end = [i for i, s in enumerate(rl) if 'End of Environment Report' in s][0]
+print("### Compile time variable report starts in line ", start)
+end = [i for i, s in enumerate(rl) if 'End of Compile-time Variable List' in s][0]
+print("### Compile time variable report ends in line   ", end-1)
+
+gams.set('gmsBOTime',[float(1960)])
+gams.set('gmsEOTime',[float(2200)])
 while start<end:
   vl = rl[start].split()
   if vl[1].lower() == 'obj':
     gams.set('gmsObj',[vl[3]])
   elif vl[1].lower() == 'botime':
     gams.set('gmsBOTime',[float(vl[3])])
+  elif vl[1].lower() == 'eotime':
+    gams.set('gmsEOTime',[float(vl[3])])
   elif vl[1].lower() == 'run_name':
     pass
   else:
     val = ' '.join(vl[3:])
-    extensions.append(('',vl[1],val,''))
+    extensions.append((vl[1],val,''))
   start += 1
 gams.set('extensions',extensions)
-$offembeddedcode TimeSlice dd MILESTONYR scenddmap gmsRunScenario offeps gmsBOTime  extensions gmsObj
-$endif.card_dd
+$offembeddedcode TimeSlice dd MILESTONYR scenddmap gmsBOTime gmsEOTime extensions gmsObj
+$endif.runmode
 
 $onEmbeddedCode Python:
 gams.wsWorkingDir = '.'
 do_print = False
 dd_db = {}
+print("\n### cube domain = " + str(os.environ['CUBEINPUTDOM']))
+domlist = str(os.environ['CUBEINPUTDOM']).split(",")
+dd_idx  = domlist.index("dd")
+reg_idx = domlist.index("ALL_REG")
+prc_idx = domlist.index("PRC")
+com_idx = domlist.index("COM_GRP")
+print("### dd_idx  = " , dd_idx)
+print("### reg_idx = " , reg_idx)
+print("### prc_idx = " , prc_idx)
+print("### com_idx = " , com_idx)
+
+print("### Turning dd files into gdx files")
 for dd in gams.get('dd'):
-  s = 'grep -iv offeps ' + r'%DDPREFIX% '.rstrip()+dd+'.dd > "' + r'%gams.scrDir%mydd.%gams.scrExt%'+'"'
+  s = 'grep -iv offeps "' + r'%DDPREFIX% '.rstrip()+dd+'.dd" > "' + r'%gams.scrDir%mydd.%gams.scrExt%'+'"'
   rc = os.system(s)
   if not rc==0:
     raise NameError('probem executing: ' + s)
```

Comment:
I can't leave a comment of the gams system call. We should run this with suppress=1 to avoid the echo print of the dd file content in the lst file

Reply:
Done in aab8650c1a739f8e5c1de9891dd7915184a285f4